test/playwright/grafana.test.ts

```diff
@@ -91,6 +91,63 @@ test("validate loki dashboard", async ({ page }) => {
   });
 });
 
+// If these tests are failing, may indicate the dashboards need to be updated
+// originally brought in from: https://github.com/keycloak/keycloak-grafana-dashboard
+test.describe("validate Keycloak Dashboards", () => {
+  const keycloakDashboards = [
+    { name: "Keycloak capacity planning dashboard", id: "keycloak-capacity-planning" },
+    { name: "Keycloak troubleshooting dashboard", id: "keycloak-troubleshooting" },
+  ];
+
+  for (const dashboard of keycloakDashboards) {
+    test(`should load ${dashboard.name} without panel errors`, async ({ page }) => {
+      const errors: string[] = [];
+
+      // Set up console error collection
+      page.on("console", msg => {
+        if (msg.type() === "error") {
+          const text = msg.text();
+          if (
+            text.includes("PanelQueryRunner Error") &&
+            !text.includes("updateAndValidate error")
+          ) {
+            errors.push(text);
+          }
+        }
+      });
+
+      // Navigate to dashboards page
+      await page.goto("/dashboards");
+
+      // Search for the dashboard
+      await page.getByPlaceholder("Search for dashboards and folders").fill("Keycloak");
+      await page.click(`text="${dashboard.name}"`);
+
+      // Wait for dashboard to load
+      await page.waitForLoadState("networkidle");
```

Comment:
Playwright docs say:
```
`'networkidle'` - **DISCOURAGED** wait until there are no network connections for at least `500` ms. Don't use
```

I know sometimes it's hard to avoid - can we wait on `domcontentloaded` instead, or a specific request that we expect? I get a bit wary of `networkidle`, especially when we're loading dashboards that might autorefresh.